src/test/Cardano/CoinSelection/Algorithm/LargestFirstSpec.hs

```diff
@@ -116,118 +127,113 @@ spec = do
-                (unsafeCoin @Int 39) (unsafeCoin @Int 43))
+            "Expect success: case #7"
+            (Right $ CoinSelectionTestResult
+                { rsInputs = [12,17,20]
+                , rsChange = [8]
+                , rsOutputs = [1,40]
+                })
             (CoinSelectionFixture
                 { maxNumOfInputs = 100
-                , utxoInputs = [12,10,17]
-                , txOutputs = [40,1,1,1]
+                , utxoInputs = [12,20,17]
+                , txOutputs = [40, 1]
                 })
 
         coinSelectionUnitTest largestFirst
-            "UTxO balance sufficient, but not fragmented enough"
-            (Left $ InputCountInsufficient $ InputCountInsufficientError 3 4)
+            "Expect success: case #8"
+            (Right $ CoinSelectionTestResult
+                { rsInputs = [10,20,20]
+                , rsChange = [3]
+                , rsOutputs = [6,41]
+                })
             (CoinSelectionFixture
                 { maxNumOfInputs = 100
-                , utxoInputs = [12,20,17]
-                , txOutputs = [40,1,1,1]
+                , utxoInputs = [20,20,10,5]
+                , txOutputs = [41, 6]
                 })
 
         coinSelectionUnitTest largestFirst
-            "UTxO balance sufficient, fragmented enough, but single output \
-            \depletes all UTxO entries"
-            (Left (InputsExhausted InputsExhaustedError))
+            "Expect success: case #9"
+            (Right $ CoinSelectionTestResult
+                { rsInputs = [6,10]
+                , rsChange = [4]
+                , rsOutputs = [1,11]
+                })
             (CoinSelectionFixture
-                { maxNumOfInputs = 100
-                , utxoInputs = [12,20,17]
-                , txOutputs = [40, 1]
+                { maxNumOfInputs = 2
+                , utxoInputs = [1,2,10,6,5]
+                , txOutputs = [11, 1]
                 })
 
         coinSelectionUnitTest largestFirst
-            "UTxO balance sufficient, fragmented enough, but single output \
-            \depletes all UTxO entries"
-            (Left (InputsExhausted InputsExhaustedError))
+            "UTxO balance not sufficient: case #1"
+            (Left $ InputValueInsufficient $ InputValueInsufficientError
+                (unsafeCoin @Int 39) (unsafeCoin @Int 40))
             (CoinSelectionFixture
                 { maxNumOfInputs = 100
-                , utxoInputs = [20,20,10,5]
-                , txOutputs = [41, 6]
+                , utxoInputs = [12,10,17]
+                , txOutputs = [40]
                 })
 
         coinSelectionUnitTest largestFirst
-            "UTxO balance sufficient, fragmented enough, but maximum input \
-            \count exceeded"
-            (Left $ InputLimitExceeded $ InputLimitExceededError 9)
+            "UTxO balance not sufficient: case #2"
+            (Left $ InputValueInsufficient $ InputValueInsufficientError
```

Comment:
this one is surprise it is error. we have to cover `[12,10,17]` and maxNumOfInputs is forgiving, and we have `40` to cover all this

Reply:
ok. I thought it was opposite inputs/utxos. nevertheless, why not to add the case I was thinking about ;-)

Reply:
> this one is surprise it is error. we have to cover `[12,10,17]` and maxNumOfInputs is forgiving, and we have `40` to cover all this

That one is actually correct, as:

* The available inputs are `[12,10,17]` (total value = `39`)
* The requested outputs are `[40,1,1,1]` (total value = `43`)

So there's not enough value in the available inputs to pay for all of the outputs.